src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleDescriptorTest.cs

```diff
@@ -8,70 +10,56 @@ namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules
     public class ConsensusRuleDescriptorTest
     {
         [Fact]
-        public void Constructor_ConsensusRuleWithValidationAttribute_InitializesClass()
+        public void Constructor_ConsensusRuleWithMempoolAndExecutionAndValidationAttribute_CannotSkipValidation()
         {
-            var rule = new TestConsensusRuleWithValidationAttribute("myRule");
+            var descriptor = new ConsensusRuleDescriptor(new RuleWithValidationExecutionMempoolAttributes());
 
-            var descriptor = new ConsensusRuleDescriptor(rule);
+            descriptor.Rule.Should().BeOfType<RuleWithValidationExecutionMempoolAttributes>();
+            descriptor.RuleAttributes.Should().HaveCount(3);
+            descriptor.RuleAttributes.Should().Contain(x => x is MempoolRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ExecutionRuleAttribute);
+            descriptor.RuleAttributes.Should().Contain(x => x is ValidationRuleAttribute);
 
-            Assert.True(descriptor.Rule is TestConsensusRuleWithValidationAttribute);
-            Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithValidationAttribute).RuleName);
-
-            Assert.Equal(3, descriptor.Attributes.Count);
-            Assert.True(descriptor.Attributes[0] is MempoolRuleAttribute);
-            Assert.True(descriptor.Attributes[1] is ExecutionRuleAttribute);
-            Assert.True(descriptor.Attributes[2] is ValidationRuleAttribute);
-
-            Assert.False(descriptor.CanSkipValidation);
+            descriptor.CanSkipValidation.Should().BeFalse();
```

Comment:
I think this check was lost in translation:
`Assert.Equal("myRule", (descriptor.Rule as TestConsensusRuleWithValidationAttribute).RuleName);`

Reply:
It looks like this field was intended to be used to make sure that the rule instance is the correct one, but it is always set to "myRule", so I actually don't see the point, and strongly typing the classes like "RuleWithMempoolAttributeButWithoutValidationAttribute" allows the same kind of checking. So I would say OK.

Reply:
no, I removed it - because it was only testing that the constructor the test double class at the bottom was setting its own field in the constructor and not the ConsensusRuleDescriptor

Reply:
Fair enough.